wallet/src/repository/wallet/tests/mod.rs

```diff
@@ -1063,3 +1072,871 @@ fn test_create_vtt_with_locked_balance() {
+    let mut output_vec: Vec<(model::OutPtr, model::OutputInfo)> = vec![];
+    for index in 0u32..1000u32 {
+        let out_pointer = model::OutPtr {
+            txn_hash: vec![0; 32],
+            output_index: index,
+        };
+        output_vec.push((
+            out_pointer,
+            model::OutputInfo {
+                pkh,
+                amount: 1,
+                time_lock: 0,
+            },
+        ));
+    }
+
+    let utxo_set: HashMap<model::OutPtr, model::OutputInfo> = HashMap::from_iter(output_vec);
+    let path = model::Path {
+        account: 0,
+        keychain: constants::EXTERNAL_KEYCHAIN,
+        index: 0,
+    };
+    let new_balance = model::BalanceInfo {
+        available: 1000u64,
+        locked: 0u64,
+    };
+    let db = HashMapDb::default();
+    db.put(&keys::account_utxo_set(0), utxo_set).unwrap();
+    db.put(&keys::account_balance(0), new_balance).unwrap();
+    db.put(&keys::pkh(&pkh), path).unwrap();
+
+    let (wallet, _db) = factories::wallet(Some(db));
+    let mut state = wallet.state.write().unwrap();
+    let request = types::DataRequestOutput {
+        witness_reward: 0,
+        witnesses: 1,
+        ..types::DataRequestOutput::default()
+    };
+
+    let value = 1000;
+    let fee = 0;
+    let err = wallet
+        .create_dr_body(&mut state, value, fee, request.clone())
+        .unwrap_err();
+
+    assert_eq!(
+        mem::discriminant(&repository::Error::MaximumDRWeightReached(request)),
+        mem::discriminant(&err),
+        "{:?}",
+        err,
+    );
+}
+
+#[test]
+fn test_create_dr_body_fee_too_large() {
+    let pkh = factories::pkh();
+    let out_pointer = model::OutPtr {
+        txn_hash: vec![0; 32],
+        output_index: 0,
+    };
+
+    let utxo_set: HashMap<model::OutPtr, model::OutputInfo> = HashMap::from_iter(vec![(
+        out_pointer,
+        model::OutputInfo {
+            pkh,
+            amount: 2000,
+            time_lock: 0,
+        },
+    )]);
+    let path = model::Path {
+        account: 0,
+        keychain: constants::EXTERNAL_KEYCHAIN,
+        index: 0,
+    };
+    let new_balance = model::BalanceInfo {
+        available: 2000u64,
+        locked: 0u64,
+    };
+    let db = HashMapDb::default();
+    db.put(&keys::account_utxo_set(0), utxo_set).unwrap();
+    db.put(&keys::account_balance(0), new_balance).unwrap();
+    db.put(&keys::pkh(&pkh), path).unwrap();
+    let (wallet, _db) = factories::wallet(Some(db));
+
+    let request = types::DataRequestOutput {
+        witness_reward: 1,
+        witnesses: 1,
+        ..types::DataRequestOutput::default()
+    };
+
+    let mut state = wallet.state.write().unwrap();
+    let value = 1;
+    let fee = u64::MAX / 2;
+    let err = wallet
+        .create_dr_body(&mut state, value, fee, request.clone())
+        .unwrap_err();
+
+    assert_eq!(
+        mem::discriminant(&repository::Error::DRFeeTooLarge(fee, request)),
+        mem::discriminant(&err),
```

Comment:
`mem::discriminant` is needed because `repository::Error` does not implement `PartialEq`. We should try to implement it as it will make our tests easier to create, but in another issue

Reply:
I will create such an issue